lib/ClangImporter/ClangImporter.cpp

```diff
@@ -951,6 +951,7 @@ ClangImporter::create(ASTContext &ctx,
   }
   auto &instance = *importer->Impl.Instance;
   instance.setInvocation(importer->Impl.Invocation);
+  instance.setVirtualFileSystem(ctx.SourceMgr.getFileSystem());
```

Comment:
Ah, hang on. This will trash any `-ivfsoverlay` options that got passed through `-Xcc`, right? For backwards compatibility, can we have this only get set if there are Swift-level `-vfsoverlay` options present? (We have access to the SearchPathOptions through the Swift ASTContext, or you could check whether the SourceManager's FileSystem has any overlays set.)

Reply:
Sounds like a comment is needed along with the fix, so folks who come by this later will have the benefit of @jrose-apple 's highi-level view.

Reply:
The plan is to declare `-vfsoverlay` incompatible with any current and future `-Xcc` VFS overlays options to avoid this ordering problem altogether and bake in the "we're nuking your overlays" behavior.  We'll warn about this with a diagnostic in the mean time - one that recommends migrating to -vfsoverlay